src/uct/rocm/copy/rocm_copy_impl.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) Advanced Micro Devices, Inc. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_ROCM_COPY_IMPL_H
+#define UCT_ROCM_COPY_IMPL_H
+
+#include <string.h>
+
+#if defined __x86_64__
+#include <cpuid.h>
+#endif
+
+#ifdef __SSE4_1__
+#include <smmintrin.h>
+
+#ifndef min
+#define min(A,B) ((A)<(B)?(A):(B))
+#endif
+
+#define _mm_load(a)    _mm_stream_load_si128((__m128i *) (a))
+#define _mm_store(a,v) _mm_storeu_si128((__m128i *) (a), (v))
+
+/* Copy data from device memory using MOVNTDQA
+ * Non temporal loads need to be 16 byte aligned
+ * Stores do not need to be aligned
+ */
+void *uct_rocm_memcpy_sse_nt_128(void *restrict dst,
```

Comment:
I think, this should go  under ucs/arch code with the rest arch and u-arch optimizations.
Generally speaking we avoid architectural ifdefs in UCP/UCT code. Corresponding  `uct_rocm_memcpy_sse_nt_128` has to be added to all architectures.

Reply:
Example: https://github.com/openucx/ucx/blob/master/src/ucs/arch/x86_64/cpu.h#L93

Reply:
You can find cpu detection code there as well.

Reply:
@shamisp Would it be better to create something like `ucs_memcpy_nontemporal()` since it is no longer specific to ROCm transport?

Reply:
@souravzzz - yes, it is good to come up with generic name. BTW @yosefe shell we move https://github.com/openucx/ucx/blob/master/src/uct/ib/mlx5/ib_mlx5.inl#L409 to arch part of the code as well. We have few ifdef in this file that potentially can be generalized.